flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobExecutionResult.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.jobmaster;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.dispatcher.Dispatcher;
+import org.apache.flink.util.SerializedThrowable;
+import org.apache.flink.util.SerializedValue;
+
+import javax.annotation.Nullable;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+import static org.apache.flink.util.Preconditions.checkArgument;
+
+/**
+ * Similar to {@link org.apache.flink.api.common.JobExecutionResult} but with an optional
+ * {@link SerializedThrowable} when the job failed.
+ *
+ * <p>This is used by the {@link JobMaster} to send the results to the {@link Dispatcher}.
+ */
+public class JobExecutionResult {
+
+	private final JobID jobId;
+
+	private final Map<String, SerializedValue<Object>> accumulatorResults;
+
+	private final long netRuntime;
+
+	private final SerializedThrowable serializedThrowable;
```

Comment:
`@Nullable` missing.

Reply:
Maybe add comment that this field is only non null if the job has failed.

Reply:
Will change it.